outbox-v2/src/main/kotlin/com/ritense/outbox/publisher/MessagePublishingFailed.kt

```diff
@@ -14,9 +14,6 @@
  * limitations under the License.
  */
 
-package com.ritense.outbox.domain
+package com.ritense.outbox.publisher
 
-import java.io.Serializable
-
-interface DomainEvent: Serializable {
-}
\ No newline at end of file
+open class MessagePublishingFailed(message: String) : RuntimeException(message)
```

Comment:
Doesn't seem to be used

Reply:
and also, adding a (optional) cause would be appreciated

Reply:
An exception class can always be added as here you have a class you can use, at it is a library / module in the end (not an end application).

I copied it from Valtimo so didn't adjust it, now did though ;) because you are right should have more options.